### PR TITLE
SOC-671 | Implement swagger in Notifications service

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/SOC-671

### CODE REVIEW

File: service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/HalLinks.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.wikia.notificationsservice.domain.responses;
+
+import lombok.Data;
+
+@Data
+public class HalLinks {
+
+  private Href __halLinks__;
```
Comment (sqreek) [inline service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/HalLinks.java]: why not halLinks?

File: service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/HalResponse.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Represents the base HAL response, will include the `_links`
+ */
+public abstract
+@Data
+class HalResponse {
+
+  @NotNull
+  private HalLinks _links;
```
Comment (sqreek) [inline service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/HalResponse.java]: why not links?

File: service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/InboxGetAllResponse.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+import java.util.List;
+
+@Data
+@EqualsAndHashCode(callSuper = false)
+public class InboxGetAllResponse extends HalResponse {
+
+  private Long count;
+  private Long total;
+  private Embedded _embedded;
```
Comment (sqreek) [inline service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/InboxGetAllResponse.java]: why not embedded?

File: service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/HalResponse.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  public HalResponse() {
```
Comment (rafalkalinski) [inline service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/HalResponse.java]: Probably this is not necessary here.

File: service/notifications/src/main/java/com/wikia/notificationsservice/resources/NotifyResource.java [hunk 5]
```diff
@@ -20,25 +26,36 @@
...
+                                  @NotNull @PathParam("targetType") String targetType,
+                                  @NotNull @PathParam("targetId") String targetId,
+                                  @NotNull @Valid Notification notification,
+                                  @Context Jedis jedis) {
     FollowableItem item = new FollowableItem(sourceType, sourceId, targetType, targetId);
 
     new QueuePublisher(rabbitConnection).publishNotificationForItem(item, notification);
 
-    return Response.status(Response.Status.CREATED).build();
+    return Response.status(Response.Status.ACCEPTED).build();
```
Comment (sqreek) [inline service/notifications/src/main/java/com/wikia/notificationsservice/resources/NotifyResource.java]: I don't see corresponding test for this change.

File: service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/InboxResponse.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package com.wikia.notificationsservice.domain.responses;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = false)
+public class InboxResponse extends HalResponse {
```
Comment (rafalkalinski) [inline service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/InboxResponse.java]: I would say that this doesn't cover how actually our response looks like. HalResponse is not necessary and properties like 'first', 'last', 'prev', etc. with HalLinks type should be declared here.

File: service/notifications/src/main/java/com/wikia/notificationsservice/resources/InboxResource.java [hunk 7]
```diff
@@ -122,9 +153,19 @@ public Response get(@PathParam("userId") Long userId,
...
+      produces = RepresentationFactory.HAL_JSON
+  )
+  @ApiResponses(value = {
+      @ApiResponse(code = 201, response = Notification.class, message = "Created"),
+      @ApiResponse(code = 500, response = Problem.class, message = "Error while inserting data")
+  })
   @POST
   @Timed
-  public Response create(@PathParam("userId") Long userId, @Valid Notification notification,
+  public Response create(@NotNull @PathParam("userId") Long userId,
```
Comment (rafalkalinski) [inline service/notifications/src/main/java/com/wikia/notificationsservice/resources/InboxResource.java]: LongParam?

File: service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/HalLinks.java (hunk 1 above)
Comment (kvas-damian) [inline service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/HalLinks.java]: fixed!

File: service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/HalResponse.java (hunk 2 above)
Comment (kvas-damian) [inline service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/HalResponse.java]: This class was used to generate swagger documentation for response. While we use HAL we should use it's naming conventions like ```_links```

File: service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/InboxGetAllResponse.java (hunk 3 above)
Comment (kvas-damian) [inline service/notifications/src/main/java/com/wikia/notificationsservice/domain/responses/InboxGetAllResponse.java]: The same reason as ```_links```

File: service/notifications/src/main/java/com/wikia/notificationsservice/resources/NotifyResource.java (hunk 5 above)
Comment (kvas-damian) [inline service/notifications/src/main/java/com/wikia/notificationsservice/resources/NotifyResource.java]: https://wikia-inc.atlassian.net/browse/SOC-596

**Comment by sqreek** [comment]
except lack of test LGTM

**Comment by rafalkalinski** [comment]
LGTM

**Pull request merged**